sklearn/preprocessing/imputation.py

```diff
@@ -374,3 +374,201 @@ def transform(self, X):
             X[coordinates] = values
 
         return X
+
+
+class MissingIndicator(BaseEstimator, TransformerMixin):
+    """Binary indicators for missing values.
+
+    Parameters
+    ----------
+    missing_values : integer or np.NaN, optional (default=np.NaN)
+        The placeholder for the missing values. All occurrences of
+        ``missing_values`` will be represented as ones.
+
+    features : {'auto' (default), 'all', array-like of int}
+        If "all", mask will represent all features.
+        If "auto", mask will only represent features with missing values
+        during fit time.
+        If mask/indices, mask will only represent features in the
+        indices or mask.
+
+    sparse : boolean or "auto", optional (default="auto")
+        If True, the transformed ``X`` will be a sparse matrix.
+        If False, the transformed ``X`` will be a numpy array.
+        If "auto", the transformed ``X`` will be of same type as input.
+
+    error_on_new : boolean, optional (default=True)
+        If True, transform will raise an error when there are features with
+        missing values in transform but have no missing values in fit
+        This is applicable only when ``features`` = "auto"
+
+    Example
+    -------
+    >>> from sklearn.preprocessing import MissingIndicator
+    >>> import numpy as np
+    >>> X1 = np.array([
+    ...       ["NaN",  1,  3],
+    ...       [ 4,  0, "NaN"],
+    ...       [ 8,  1,  0]
+    ... ])
+    >>> X2 = np.array([
+    ...       [ 5, 1, "NaN"],
+    ...       ["NaN",  2,  3],
+    ...       [ 2,  4,  0]
+    ... ])
+    >>> indicator = MissingIndicator()
+    >>> indicator.fit(X1)
+    MissingIndicator(error_on_new=True, features='auto', missing_values='NaN',
+             sparse='auto')
+    >>> X2_tr = indicator.transform(X2)
+    >>> X2_tr
+    array([[0, 1],
+           [1, 0],
+           [0, 0]])
+
+    Attributes
+    ----------
+    feat_with_missing_  : array of shape(n_missing_features,)
```

Comment:
space after shape.

Reply:
I think this should just be called `features_` as an estimated version of `features`.